protocol/x/sending/e2e/isolated_subaccount_transfers_test.go

```diff
@@ -0,0 +1,394 @@
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 10_000_000_000, // $10,000 USDC
+			},
+			senderSubaccountId:   constants.Bob_Num0,
+			receiverSubaccountId: constants.Alice_Num0,
+			quantums:             100_000_000, // $100
+			liquidityTiers:       constants.LiquidityTiers,
+			perpetuals: []perptypes.Perpetual{
+				constants.BtcUsd_20PercentInitial_10PercentMaintenance,
+				constants.EthUsd_20PercentInitial_10PercentMaintenance,
+				constants.IsoUsd_IsolatedMarket,
+			},
+			expectedSubaccounts: []satypes.Subaccount{
+				changeUsdcBalance(constants.Alice_Num0_1ISO_LONG_10_000USD, 100_000_000),
+				changeUsdcBalance(constants.Bob_Num0_10_000USD, -100_000_000),
+			},
+			expectedCollateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 9_900_000_000, // $9,900 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 10_100_000_000, // $10,100 USDC
+			},
+			expectedErr: "",
+		},
+		`Can transfer from isolated subaccount to isolated subaccount in different isolated markets, and
+		coins are sent from one isolated collateral pool to the other isolated collateral pool`: {
+			subaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_1ISO_LONG_10_000USD,
+				constants.Bob_Num0_1ISO2_LONG_10_000USD,
+			},
+			collateralPoolBalances: map[string]int64{
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.Iso2Usd_IsolatedMarket.Params.Id),
+				).String(): 10_000_000_000, // $10,000 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 10_000_000_000, // $10,000 USDC
+			},
+			senderSubaccountId:   constants.Alice_Num0,
+			receiverSubaccountId: constants.Bob_Num0,
+			quantums:             100_000_000, // $100
+			liquidityTiers:       constants.LiquidityTiers,
+			perpetuals: []perptypes.Perpetual{
+				constants.BtcUsd_20PercentInitial_10PercentMaintenance,
+				constants.EthUsd_20PercentInitial_10PercentMaintenance,
+				constants.IsoUsd_IsolatedMarket,
+				constants.Iso2Usd_IsolatedMarket,
+			},
+			expectedSubaccounts: []satypes.Subaccount{
+				changeUsdcBalance(constants.Alice_Num0_1ISO_LONG_10_000USD, -100_000_000),
+				changeUsdcBalance(constants.Bob_Num0_1ISO2_LONG_10_000USD, 100_000_000),
+			},
+			expectedCollateralPoolBalances: map[string]int64{
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 9_900_000_000, // $9,900 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.Iso2Usd_IsolatedMarket.Params.Id),
+				).String(): 10_100_000_000, // $10,100 USDC
+			},
+			expectedErr: "",
+		},
```

Comment:
Another possible test case?
- collateral transfer from isolated subaccount to isolated subaccount with the same isolated perpetual, doesn't result in underlying transfer of collateral pool

Reply:
Good idea, added.